test/topology_custom/test_maintenance_mode.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from cassandra.protocol import ConfigurationException
+from cassandra.connection import UnixSocketEndPoint
+from cassandra.cluster import Cluster
+from cassandra.policies import WhiteListRoundRobinPolicy
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+import logging
+import socket
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.asyncio
+async def test_maintenance_mode(manager: ManagerClient):
+    """
+    The test checks that in maintenance mode server A is not available for other nodes and for clients.
+    It is possible to connect by the maintenance socket to server A and perform local CQL operations.
+    """
+
+    server_a, server_b = await manager.server_add(), await manager.server_add()
+    workdir = await manager.server_get_workdir(server_a.server_id)
+    socket_path = workdir + "/cql.m"
+
+    cluster = Cluster([server_b.ip_addr])
+    cql = cluster.connect()
```

Comment:
> Why use a separate cql session instead of doing what other tests

I wanted to be consistent with the rest of this test, but it's not significant.

Reply:
Ok I agree that consistency is important.

However note that with `manager.cql` and `run_async` we have custom timeout options, which we set after observing flakiness in debug mode due to queries taking a long time sometimes.

This is the function we use to create the `manager.cql` session:
```python3
def cluster_con(hosts: List[IPAddress], port: int, use_ssl: bool, auth_provider=None):
    """Create a CQL Cluster connection object according to configuration.
       It does not .connect() yet."""
    assert len(hosts) > 0, "python driver connection needs at least one host to connect to"
    profile = ExecutionProfile(
        load_balancing_policy=RoundRobinPolicy(),
        consistency_level=ConsistencyLevel.LOCAL_QUORUM,
        serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL,
        # The default timeouts should have been more than enough, but in some
        # extreme cases with a very slow debug build running on a slow or very busy
        # machine, they may not be. Observed tests reach 160 seconds. So it's
        # incremented to 200 seconds.
        # See issue #11289.
        # NOTE: request_timeout is the main cause of timeouts, even if logs say heartbeat
        request_timeout=200)
    whitelist_profile = ExecutionProfile(
        load_balancing_policy=TokenAwarePolicy(WhiteListRoundRobinPolicy(hosts)),
        consistency_level=ConsistencyLevel.LOCAL_QUORUM,
        serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL,
        request_timeout=200)
    if use_ssl:
        # Scylla does not support any earlier TLS protocol. If you try,
        # you will get mysterious EOF errors (see issue #6971) :-(
        ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
    else:
        ssl_context = None

    return Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile, 'whitelist': whitelist_profile},
                   contact_points=hosts,
                   port=port,
                   # TODO: make the protocol version an option, to allow testing with
                   # different versions. If we drop this setting completely, it will
                   # mean pick the latest version supported by the client and the server.
                   protocol_version=4,
                   # NOTE: No auth provider as auth keysppace has RF=1 and topology will take
                   # down nodes, causing errors. If auth is needed in the future for topology
                   # tests, they should bump up auth RF and run repair.
                   ssl_context=ssl_context,
                   # The default timeouts should have been more than enough, but in some
                   # extreme cases with a very slow debug build running on a slow or very busy
                   # machine, they may not be. Observed tests reach 160 seconds. So it's
                   # incremented to 200 seconds.
                   # See issue #11289.
                   connect_timeout = 200,
                   control_connection_timeout = 200,
                   # NOTE: max_schema_agreement_wait must be 2x or 3x smaller than request_timeout
                   # else the driver can't handle a server being down
                   max_schema_agreement_wait=20,
                   idle_heartbeat_timeout=200,
                   # The default reconnection policy has a large maximum interval
                   # between retries (600 seconds). In tests that restart/replace nodes,
                   # where a node can be unavailable for an extended period of time,
                   # this can cause the reconnection retry interval to get very large,
                   # longer than a test timeout.
                   reconnection_policy = ExponentialReconnectionPolicy(1.0, 4.0),

                   auth_provider=auth_provider,
                   # Capture messages for debugging purposes.
                   connection_class=CustomConnection
                   )

```

from test/topology/conftest.py

Note the custom timeout options.


We also have:
```python3
# TODO: paged result query handling (iterable?)
def run_async(self, *args, all_pages = False, **kwargs) -> asyncio.Future:
    """Execute a CQL query asynchronously by wrapping the driver's future"""
    # The default timeouts should have been more than enough, but in some
    # extreme cases with a very slow debug build running on a slow or very busy
    # machine, they may not be. Observed tests reach 160 seconds. So it's
    # incremented to 200 seconds.
    # See issue #11289.
    kwargs.setdefault("timeout", 200.0)
    return _wrap_future(self.execute_async(*args, **kwargs), all_pages = all_pages)
```

in test/pylib/async_cql.py

You should use the same settings in custom cql sessions too.
Maybe you could somehow generalize `cluster_con` and then reuse it in the test.
And you should be able to use `run_async` on any session.

Reply:
In `test/topology/conftest.py` we have
```python
Session.run_async = run_async     # patch Session for convenience
```
so every Session has this method and `cql.execute` can be easily replaced with `await cql.run_async` wiithout any change in the conftest file.

Reply:
Ah, you mean also execution profile, nvm